src/project_manager/ros_project_plugin.cpp

```diff
@@ -159,6 +171,39 @@ void ROSProjectPlugin::createCppCodeStyle()
   originalCppCodeStylePreferences->fromSettings(QLatin1String(CppTools::Constants::CPP_SETTINGS_ID), s);
 }
 
+void ROSProjectPlugin::reloadProjectBuildInfo()
+{
+    QFutureInterface<void> f;
+
+    f.setProgressRange(0, 100);
+    Core::ProgressManager::addTask(f.future(),
+                                   tr("Reloading Project Build Info"),
+                                   Constants::ROS_RELOADING_BUILD_INFO);
+
+    f.reportStarted();
+    ROSProject *rosProject = qobject_cast<ROSProject *>(ProjectTree::currentProject());
+    if (rosProject) {
+        // Rebuild code model
+        rosProject->refreshCppCodeModel();
+        f.setProgressValue(50);
+
+        // Update environment
+        ROSBuildConfiguration *bc = rosProject->rosBuildConfiguration();
+        if (bc) {
+            ROSUtils::WorkspaceInfo workspaceInfo = ROSUtils::getWorkspaceInfo(bc->project()->projectDirectory(), bc->buildSystem(), bc->project()->distribution());
+            Utils::Environment env(ROSUtils::getWorkspaceEnvironment(workspaceInfo).toStringList());
+            bc->updateQtEnvironment(env);
+
+            f.setProgressValue(100);
+            f.reportFinished();
+            return;
+        }
+    }
+    f.reportCanceled();
+    f.reportFinished();
```

Comment:
Yea, it appears so. I originally tested it where I just called reportCanceled and the progress bar stayed open and never closed. When I added the reportFinished the progress went to 100% then turned red and eventually faded away. Not sure if this is required by the future or if this has to due with the progress manager.